smithy-swift-codegen/src/test/kotlin/serde/xml/UnionEncodeXMLGenerationTests.kt

```diff
@@ -65,93 +65,78 @@ class UnionEncodeXMLGenerationTests {
-                    if let structvalue = structvalueDecoded {
-                        self = .structvalue(structvalue)
-                        return
-                    }
-                    if containerValues.contains(.mapvalue) {
-                        struct KeyVal0{struct K{}; struct V{}}
-                        let mapvalueWrappedContainer = containerValues.nestedContainerNonThrowable(keyedBy: ClientRuntime.MapEntry<Swift.String, Swift.String, KeyVal0.K, KeyVal0.V>.CodingKeys.self, forKey: .mapvalue)
-                        if let mapvalueWrappedContainer = mapvalueWrappedContainer {
-                            let mapvalueContainer = try mapvalueWrappedContainer.decodeIfPresent([ClientRuntime.MapKeyValue<Swift.String, Swift.String, KeyVal0.K, KeyVal0.V>].self, forKey: .entry)
-                            var mapvalueBuffer: [Swift.String:Swift.String]? = nil
-                            if let mapvalueContainer = mapvalueContainer {
-                                mapvalueBuffer = [Swift.String:Swift.String]()
-                                for stringContainer0 in mapvalueContainer {
-                                    mapvalueBuffer?[stringContainer0.key] = stringContainer0.value
+                    let key = containerValues.allKeys.first
+                    switch key {
+                        case .doublevalue:
+                            let doublevalueDecoded = try containerValues.decode(Swift.Double.self, forKey: .doublevalue)
+                            self = .doublevalue(doublevalueDecoded)
+                        case .datavalue:
+                            if containerValues.contains(.datavalue) {
+                                do {
+                                    let datavalueDecoded = try containerValues.decodeIfPresent(ClientRuntime.Data.self, forKey: .datavalue)
+                                    self = .datavalue(datavalueDecoded)
+                                } catch {
+                                    self = .datavalue("".data(using: .utf8))
                                 }
+                            } else {
+                                //No-op
                             }
-                            if let mapvalue = mapvalueBuffer {
-                                self = .mapvalue(mapvalue)
-                                return
+                        case .unionvalue:
+                            let unionvalueDecoded = try containerValues.decode(Box<RestXmlProtocolClientTypes.XmlUnionShape>.self, forKey: .unionvalue)
+                            self = .unionvalue(unionvalueDecoded.value)
+                        case .structvalue:
+                            let structvalueDecoded = try containerValues.decode(RestXmlProtocolClientTypes.XmlNestedUnionStruct.self, forKey: .structvalue)
+                            self = .structvalue(structvalueDecoded)
+                        case .mapvalue:
+                            if containerValues.contains(.mapvalue) {
```

Comment:
not sure if we need these if statements anymore but didn't want to remove just yet until I checked with you @wooj2

Reply:
resolved offline., resolving.